pkg/recipes/error.go

```diff
@@ -50,11 +53,29 @@ func NewRecipeError(code string, message string, deploymentStatus util.RecipeDep
 	return err
 }
 
-// GetRecipeErrorDetails is used to get ErrorDetails if error is of type RecipeError else returns nil.
-func GetRecipeErrorDetails(err error) *v1.ErrorDetails {
-	recipeError, _ := err.(*RecipeError)
-	if recipeError != nil {
-		return &recipeError.ErrorDetails
+// GetErrorDetails is used to get ErrorDetails from different error types.
+func GetErrorDetails(err error) *v1.ErrorDetails {
+	if err == nil {
+		return nil
+	}
+
+	switch v := err.(type) {
+	case *RecipeError:
+		return &v.ErrorDetails
+
+	case *azcore.ResponseError:
+		data := v1.ErrorResponse{}
```

Comment:
Should we use https://github.com/radius-project/radius/blob/main/pkg/azure/clientv2/unfold.go#L171 here?

Reply:
Thanks for pointing this @kachawla , I will update it to use unfold.